pyemu/pst/pst_handler.py

```diff
@@ -2585,11 +2585,14 @@ def adjust_weights(self, obs_dict=None, obsgrp_dict=None):
         if obsgrp_dict is not None:
             obs = self.observation_data
             # first zero-weight all obs in groups specified to have 0 contrib to phi
+            zero_weight_groups = []
             for grp, contrib in obsgrp_dict.items():
                 if contrib==0:
                     obs.loc[obs.obgnme == grp, "weight"] = 0.0
                     # drop zero- contribution groups
-                    del obsgrp_dict[grp]
+                    zero_weight_groups.append(grp)
+            obsgrp_dict = {k: v for k, v in obsgrp_dict.items() 
+                           if k not in zero_weight_groups}
             # reset groups with all zero weights
             for grp in obsgrp_dict.keys():
                 if obs.loc[obs.obgnme == grp, "weight"].sum() == 0.0:
```

Comment:
I wonder if we can combine these loops in to one (and avoid that runtime dict size change error) by looping over `list(obsgrp_dict.keys())`.
So:
```
for grp in list(obsgrp_dict.keys()):
    if obsgrp_dict[grp]==0:
        obs.loc[obs.obgnme==grp,"weight"]=0
        del obsgrp_dict[grp]
    elif obs.loc[obs.obgnme==grp,"weight"].sum()==0:
        obs.loc[obs.obgnme==grp,"weight"]=1
```